src/vs/code/electron-main/app.ts

```diff
@@ -225,12 +225,15 @@ export class CodeApplication extends Disposable {
 				this.nativeHostMainService?.openExternal(undefined, url);
 			});
 
-			session.defaultSession.setPermissionRequestHandler((webContents, permission /* 'media' | 'geolocation' | 'notifications' | 'midiSysex' | 'pointerLock' | 'fullscreen' | 'openExternal' */, callback) => {
+			session.defaultSession.setPermissionRequestHandler((_webContents, permission /* 'media' | 'geolocation' | 'notifications' | 'midiSysex' | 'pointerLock' | 'fullscreen' | 'openExternal' */, callback, details) => {
+				if (permission === 'clipboard-read') {
```

Comment:
I believe I can gate this check to only enable clipboard access inside webviews. The basic approach would be to make sure the requestingUrl looks like it comes from a webview.

For a more complete fix though, I'd need to delegate this check into the main thread webview service to check that the requesting url comes from an webview that we know exists. This may be overkill though, so I left it out of the first version of this PR

Reply:
Don't webview use a separate session than the workbench, why do we need to perform this check on the session of workbench ?

Reply:
This is for iframe based webviews. As far as I know, there is no way to run these in their own session. Please let me know if I am wrong

Reply:
Ah didn't remember that, looks good then.

> I believe I can gate this check to only enable clipboard access inside webviews. The basic approach would be to make sure the requestingUrl looks like it comes from a webview.

This would be a good thing to do in a follow-up